test/test_helpers/httptestserver.cpp

```diff
@@ -0,0 +1,91 @@
+#include "httptestserver.h"
+
+#include <stdexcept>
+#include <string>
+
+namespace test_helpers {
+
+static HttpTestServer* instance = nullptr;
+
+HttpTestServer::HttpTestServer()
+	: process({"./http-test-server"})
+{
+	address = process.readLine();
+	instance = this;
+}
+
+HttpTestServer::~HttpTestServer()
+{
+	process.writeLine("exit");
+	process.waitForExit();
+	instance = nullptr;
+}
+
+HttpTestServer& HttpTestServer::getInstance()
+{
+	if (instance == nullptr) {
+		throw std::runtime_error("No HttpTestServer instantiated");
+	}
+	return *instance;
+}
```

Comment:
Why is this implemented differently from how singleton pattern is usually implemented in C++ (a `static` variable inside a static method + making all constructors private)? Do we somehow rely on the fact that `HttpTestServer` is instantiated after the calls to `setlocale` and `srand` in `main`?

Reply:
I've mainly implemented it this way to keep full control of when it is constructed (before running any tests) and when it is destructed (to make it easier to debug cleanup issues).

I just tried to use the "static variable inside a static method" way and ran into issues with the test returning exit code 141.
Not sure why.
I prefer to leave the implementation as-is.
However, if you want, I can look into it further (just not today).